pipelines/tekton/gitops-update-pipeline/gitops-update-pipeline.yaml

```diff
@@ -0,0 +1,127 @@
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: gitops-update-pipeline
+spec:
+  params:
+    - name: gitServer
+      type: string
+    - name: gitOrgName
+      type: string
+    - name: gitRepoName
+      type: string
+    - name: imageDigest
+      type: string
+    - name: imageReferenceFilePath
+      type: string
+    - name: gitApiPrefix
+      default: /api/v1
+      type: string
+    - name: gitTokenSecretName
+      type: string
+    - name: gitTokenSecretKey
+      default: token
+      type: string
+  tasks:
+    - name: git-clone
+      params:
+        - name: url
+          value: $(params.gitServer)/$(params.gitOrgName)/$(params.gitRepoName)
+        - name: gitInitImage
+          value: registry.redhat.io/openshift-pipelines/pipelines-git-init-rhel8@sha256:1a50511583fc02a27012d17d942e247813404104ddd282d7e26f99765174392c
+        - name: subdirectory
+          value: /$(params.gitRepoName)/
+      taskRef:
+        kind: ClusterTask
+        name: git-clone
+      workspaces:
+        - name: output
+          workspace: git-workspace
+        - name: basic-auth
+          workspace: git-basic-auth
+    - name: commit-image-ref
+      params:
+        - name: BASE_IMAGE
+          value: cgr.dev/chainguard/git:root-2.39@sha256:7759f87050dd8bacabe61354d75ccd7f864d6b6f8ec42697db7159eccd491139
+        - name: GIT_USER_NAME
+          value: Pipeline Image Updater
+        - name: GIT_USER_EMAIL
+          value: pipeline@example.com
+        - name: GIT_SCRIPT
+          value: |
+            # Move repo to HOME so Git does not complain
+            # https://stackoverflow.com/a/71941707/19020549
+            cp -r $(workspaces.source.path)/$(params.gitRepoName) ~/$(params.gitRepoName)
+
+            cd ~/$(params.gitRepoName)
+
+            PR_BRANCH=pipeline_$(context.pipelineRun.uid)
+            git checkout -b $PR_BRANCH
+
+            sed -i -e "s/digest: .*/digest: $(params.imageDigest)/" $(params.imageReferenceFilePath)
+
+            if [ -z "$(git status --porcelain)" ]; then
+                echo "Update did not cause a modification"
+                exit 0
+            fi
+
+            git commit -am "Update image ref from pipeline"
+            git push origin $PR_BRANCH:$PR_BRANCH
```

Comment:
Do you think we should bother moving this to a ConfigMap?

Reply:
This is a generic git workflow so I don't have any issues leaving this embedded in the pipeline definition.  There will be some future cleanup of the pipelines but, IMO, I have no major concerns with this implementation right now